cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -416,7 +437,46 @@ public void getApplicationDetails(FullHttpRequest request, HttpResponder respond
   }
 
   /**
-   * Decodes request coming from the {@link #getApplicationDetails(FullHttpRequest, HttpResponder, String)} call.
+   * Get {@link ApplicationDetail} for a given application
+   *
+   * @param request     {@link HttpRequest}
+   * @param responder   {@link HttpResponse}
+   * @param namespace   the namespace to get all application details   *
+   * @param application the id of the application to get its {@link ApplicationDetail}
+   * @throws Exception if either namespace or application doesn't exist, or failed to get {@link ApplicationDetail}
+   */
+  @GET
+  @Path("/appdetail/{app-id}")
+  public void getAppDetail(HttpRequest request, HttpResponder responder,
```

Comment:
why do we need this endpoint? Isn't it the same as apps/app-d?

Reply:
Yes. It is the same as /app/<appid>. I hope at some point we can change /app/<appid> to be consistent with /apps to return ApplicationRecord instead of ApplicationDetail. 

Introduce this /appdetail/<appid> to achieve the parity, as I dislike the fact that client uses /app/<appid> in get-one case and /appdetail for get-all case.

Reply:
we should not have duplicate endpoints in this class.

I think it's ok if there is an internal handler that has this endpoint, but it shouldn't be mixed in with the public APIs.

Reply:
Done. Moved to internal endpoint